lib/config.py

```diff
@@ -126,6 +138,26 @@ def parse_options(type):
     for key, credentials in _get_parameters_credentials().items():
         for name, description in credentials.items():
             parser.add_argument('--{}'.format(name), help=description)
+
+    return parser
+
+def parse_options(type):
+    """Parse the required command line options for the given operation type.
+
+    :param type: a string containing either `backup` or `restore`
+    :returns: a dictionary containing the key-value pairs of the provided parameters
+
+    :Example:
+        ::
+
+            configuration = parse_options('backup')
+    """
+
+    # first, remove all the old logs and data
+    remove_old_logs_state()
```

Comment:
I agree. Will update the PR.

Reply:
It can be renamed as 'preProcessOperation' and do the same thing. Deletion of log can be exposed for caller. But let's do a default cleanup also, in order to not make client code have too many steps to adapt the library.

Reply:
@sauravmndl @ketakig12 Apologies for taking up this again after long time. So to accommodate both above comments, I am planning following:
1. Will keep the previous log cleanup as a default step, but will write to services teams regarding this and take acknowledgement from them that default cleanup is not a problem to them.
2. Changing the name of function will need services teams to modify their codebase, which would be unnecessary if default cleanup is not problematic to them. I will, however, modify the documentation of this library to clearly state that default cleanup will be done as part of `parse_options` functions.

Hope this is fine. Please let me know if there are concerns with this plan.

Reply:
@jayeshjk Okay.